### PR TITLE
Add colocalization measures

### PR DESCRIPTION
**Pull request by grlee77**
This MR impelements four colocaliziation measures being introduced for scikit-image 0.20. The implementation of each is fairly simple. For reference, see a related [gallery example](https://scikit-image.org/docs/dev/auto_examples/applications/plot_colocalization_metrics.html#sphx-glr-auto-examples-applications-plot-colocalization-metrics-py).

## Benchmark results vs. scikit-image 0.20dev

Benchmark results show speedup in all cases below aside from `intersection_coeff` on small images.

function              | shape         | dtype   | acceleration
----------------------|---------------|---------|-------------
manders_coloc_coeff   | (512, 512)    | float32 |  1.72
manders_coloc_coeff   | (512, 512)    | uint8   |  2.93
manders_coloc_coeff   | (3840, 2160)  | float32 | 39.3
manders_coloc_coeff   | (3840, 2160)  | uint8   | 49.9
manders_coloc_coeff   | (192,192,192) | float32 | 38.1
manders_coloc_coeff   | (192,192,192) | uint8   | 47.4

function              | shape         | dtype   | acceleration
----------------------|---------------|---------|-------------
manders_overlap_coeff | (512, 512)    | float32 |  1.41
manders_overlap_coeff | (512, 512)    | uint8   |  1.48
manders_overlap_coeff | (3840, 2160)  | float32 |  2.33
manders_overlap_coeff | (3840, 2160)  | uint8   |  2.93
manders_overlap_coeff | (192,192,192) | float32 |  2.31
manders_overlap_coeff | (192,192,192) | uint8   |  2.80

function              | shape         | dtype   | acceleration
----------------------|---------------|---------|-------------
pearson_corr_coeff    | (512, 512)    | float32 |  6.96
pearson_corr_coeff    | (512, 512)    | uint8   |  8.10
pearson_corr_coeff    | (3840, 2160)  | float32 | 49.5
pearson_corr_coeff    | (3840, 2160)  | uint8   | 53.4
pearson_corr_coeff    | (192,192,192) | float32 | 51.6
pearson_corr_coeff    | (192,192,192) | uint8   | 52.5

function              | shape         | dtype   | acceleration
----------------------|---------------|---------|-------------
intersection_coeff    | (512, 512)    | bool    |  0.278
intersection_coeff    | (3840, 2160)  | bool    | 13.1
intersection_coeff    | (192,192,192) | bool    | 11.1

### CODE REVIEW

**Comment by gigony** [comment]
intersection_coeff	(512, 512)	bool	0.278

I wonder if this result is okay and if you have an idea on what caused the performance degradation.

File: python/cucim/src/cucim/skimage/measure/_colocalization.py [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+    ----------
+    image0 : (M, N) ndarray
+        Image of channel A. All pixel values should be non-negative.
+    image1 : (M, N) ndarray
+        Image of channel B. All pixel values should be non-negative.
+        Must have same dimensions as `image0`
+    mask : (M, N) ndarray of dtype bool, optional
+        Only `image0` and `image1` pixel values within this region of interest
+        mask are included in the calculation.
+        Must have ♣same dimensions as `image0`.
```
Comment (gigony) [inline python/cucim/src/cucim/skimage/measure/_colocalization.py:198]:
```suggestion
        Must have same dimensions as `image0`.
```

**Review by gigony — APPROVED** [review]
Thanks @grlee77 ! It looks good to me! (with a minor typo that can be addressed later).

File: python/cucim/src/cucim/skimage/measure/_colocalization.py (hunk 1 above)
Comment (grlee77) [reply python/cucim/src/cucim/skimage/measure/_colocalization.py:198]: Thanks, not sure what happened there!

**Pull request merged**